placement/algo/helper.go

```diff
@@ -241,11 +243,14 @@ func (ph placementHelper) canAssignRack(shard uint32, from placement.HostShards,
 	return true
 }
 
-func (ph placementHelper) canAssignHost(shard uint32, from, to placement.HostShards) bool {
+func (ph placementHelper) canAssignHost(shard uint32, from, to placement.HostShards, rackCheck bool) bool {
```

Comment:
yeah if the placementHelper stores the placement option inside, it doesn't need the `rackCheck` flag to be passed in.